<!DOCTYPE html>
<html lang="en">
  <head>

    <meta charset="utf-8">
    <title>Scripts for structural biology</title>
    <meta name="description" content="Scripts for structural biology">
    <meta name="author"      content="Katie Edmonds">
    <meta name="keywords"    content="pymol, linux, all">
    <meta name="viewport"    content="width=device-width, initial-scale=1.0">
    <link rel="icon"      href="/favicon.ico">
    <link rel="canonical" href="http://www.example.com/index.html">


    <!-- CSS -->
    <link rel="stylesheet" type="text/css" href="/css/bootstrap.min.css">
    <link rel="stylesheet" type="text/css" href="/css/pygments.css">
    <link rel="stylesheet" type="text/css" href="/css/scribble.css">
    <link rel="stylesheet" type="text/css" href="/css/custom.css">
    <!-- Feeds -->
    <link ref="alternate" type="application/atom+xml"
          href="/feeds/all.atom.xml" title="Atom Feed">
    <link ref="alternate" type="application/rss+xml"
          href="/feeds/all.rss.xml" title="RSS Feed">
    <!-- JS -->
    <script type="text/javascript">
      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-xxxxx']);
      _gaq.push(['_setDomainName', 'example.com']);
      _gaq.push(['_trackPageview']);
      setTimeout(function(){_gaq.push(['_trackEvent', '30_seconds', 'read'])}, 30000); // http://drawingablank.me/blog/fix-your-bounce-rate.html
      (function() {
          var ga = document.createElement('script');
          ga.type = 'text/javascript'; ga.async = true;
          ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
          var s = document.getElementsByTagName('script')[0];
          s.parentNode.insertBefore(ga, s);
      })();
    </script>
  </head>
  <body>
    <!-- A standard Twitter Bootstrap nav bar -->
    <header class="navbar navbar-default navbar-inverse"
            role="banner">
      <div class="container">
        <div class="navbar-header">
          <button type="button"
                  class="navbar-toggle"
                  data-toggle="collapse"
                  data-target=".our-nav-collapse">
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a href="/index.html" class="navbar-brand">My Blog</a>
        </div>
        <div class="collapse navbar-collapse our-nav-collapse"
             role="navigation">
          <ul class="nav navbar-nav">

            <li class="dropdown">
              <a href="#" class="dropdown-toggle" data-toggle="dropdown">
                Tags <b class="caret"></b></a>
              <ul class="dropdown-menu">
                <li><a href="/index.html">All Posts</a></li>

<li><a href="/tags/linux.html">linux</a></li>

<li><a href="/tags/pymol.html">pymol</a></li>
              </ul>
            </li>
            <li>
              <a href="/About.html">About</a>
            </li> 
            <li><a href="/feeds/all.atom.xml">Atom</a></li>
            <li><a href="/feeds/all.rss.xml">RSS</a></li>
          </ul>
        </div>
      </div>
    </header>
    <div class="container">
      <div class="row">

        <!-- Main column -->
        <div id="content" class="col-md-12">

          <h1>Welcome</h1>
          <p>This blog is devoted to little tricks for structural biology.</p> 



          <article>
  <header>
    <h2><a href='/blog/2014/12/19/a-few-useful-bioinformatics-sites/'>A few useful bioinformatics sites</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-12-19" pubdate="true">2014-12-19</time> :: <span class="tags"><a href="/tags/.html"></a></span></p>
  </header>

<ul>
 <li><a href="http://emboss.bioinformatics.nl/cgi-bin/emboss/palindrome">EMBOSS palindrome finder</a> is the best tool I&rsquo;ve been able to find for searching for transcription factor binding sites that consist of inverted repeats separated by a gap. It&rsquo;s still tricky to decide on minimum length, gap size, and number of mismatches to get an interpretable set of results.</li>
 <li><a href="http://rsat.ulb.ac.be/rsat/retrieve-seq_form.cgi">RSAT</a> is useful for fetching intergenic regions upstream or downstream of a gene of interest, but can also be used for fetching the ORF. For example, you might be interested in the sequence upstream of spr1672 to look for regulatory elements. The default parameter is to fetch the whole region upstream of the gene, but you can tune it in other ways. <a href="http://rsat.ulb.ac.be/rsat/">The parent site</a> has lots of other tools that are worth exploring.</li>
 <li><a href="http://espript.ibcp.fr/ESPript/ESPript/">ESPript3</a> is a great tool for making your multiple sequence alignment look pretty, especially if you want to display secondary structure elements from a PDB file across the top.</li>
 <li><a href="http://espript.ibcp.fr/ESPript/cgi-bin/ENDscript.cgi">ENDscript2</a> is a great tool for a quick and dirty initial look at a PDB entry. It searches the PDB for homologous structures, aligns them, and displays the multiple sequence alignment with the secondary structure elements, solvent accessibility, etc, and gives you pymol scripts or session files showing sequence and structure conservation.</li>
 <li>There are a lot of approaches to homology modelling. I like BLASTing against the PDB and finding my favorite homologous protein, then going to MODELLER in the <a href="http://toolkit.tuebingen.mpg.de/">bioinformatics toolkit</a>, which also has many other useful tools. Sometimes it&rsquo;s painfully picky about silly details of the input format, but it has given me some nice looking results.</li>
 <li>
  <p>Do you wish you could restrict BLAST results to those genes or organisms that have an NCBI Gene entry that will allow you to explore the genomic context? So do I. I haven&rsquo;t found the answer yet, but restricting a search to organisms listed <a href="http://www.ncbi.nlm.nih.gov/genome/browse/representative/">here</a> as representative genomes tends to help. Getting all the tax_ids from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/TaxIdentifier/tax_identifier.cgi">here</a> can help. This permits searching with an unweildy Entrez Query like</p>
  <pre><code>txid208435 [ORGN] OR txid862971 [ORGN] OR txid888833 [ORGN] OR txid1123298 [ORGN] OR txid482234 [ORGN] OR txid862969 [ORGN] OR txid873449 [ORGN] OR txid889201 [ORGN] OR txid1123301 [ORGN] OR txid904293 [ORGN] OR txid486410 [ORGN] OR txid1123302 [ORGN] OR txid552526 [ORGN] OR txid864569 [ORGN] OR txid1123303 [ORGN] OR txid637909 [ORGN] OR txid467705 [ORGN] OR txid1123304 [ORGN] OR txid764299 [ORGN] OR txid471872 [ORGN] OR txid1069533 [ORGN] OR txid68892 [ORGN] OR txid871237 [ORGN] OR txid1318633 [ORGN] OR txid591365 [ORGN] OR txid1076934 [ORGN] OR txid764298 [ORGN] OR txid1116231 [ORGN] OR txid1123306 [ORGN] OR txid1123307 [ORGN] OR txid1123308 [ORGN] OR txid1123309 [ORGN] OR txid210007 [ORGN] OR txid1302863 [ORGN] OR txid927666 [ORGN] OR txid1123311 [ORGN] OR txid1123312 [ORGN] OR txid760570 [ORGN] OR txid936154 [ORGN] OR txid981540 [ORGN] OR txid888746 [ORGN] OR txid1123313 [ORGN] OR txid171101 [ORGN] OR txid170187 [ORGN] OR txid873448 [ORGN] OR txid1054460 [ORGN] OR txid373153 [ORGN] OR txid910313 [ORGN] OR txid160490 [ORGN] OR txid699248 [ORGN] OR txid347253 [ORGN] OR txid388919 [ORGN] OR txid1123317 [ORGN] OR txid1074052 [ORGN] OR txid1156433 [ORGN] OR txid391295 [ORGN] OR txid568814 [ORGN] OR txid299768 [ORGN] OR txid1123318 [ORGN] OR txid1282664 [ORGN] OR txid218495 [ORGN] OR txid764291 [ORGN] OR txid904306 [ORGN] OR txid365659 [ORGN]</code></pre></li></ul>
<footer class="read-more"><a href="/blog/2014/12/19/a-few-useful-bioinformatics-sites/">&hellip;more&hellip;</a></footer>
</article>
<article>
  <header>
    <h2><a href='/blog/2014/12/18/pymol-sphere-representation-and-ionic-radii/'>Pymol: sphere representation and ionic radii</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-12-18" pubdate="true">2014-12-18</time> :: <span class="tags"><a href="/tags/pymol.html">pymol</a></span></p>
  </header>

<p>I just want to point out and reiterate an old <a href="https://www.mail-archive.com/pymol-users@lists.sourceforge.net/msg04718.html">forum message</a> about using PyMOL to display ions as spheres. The spheres representation shows the van der Waals radius of an atom. Apparently PyMOL stores the nonionic radii by default, 1.73 A for Mg, for example, rather than 0.71 A for Mg++. If you look up the desired ionic radius:</p>

<p><a href="https://www.webelements.com/nickel/atom_sizes.html">https://www.webelements.com/nickel/atom_sizes.html</a></p>

<p>You can set them in PyMOL using the alter command, to obtain a more appropriate picture:</p>

<pre><code> PyMOL&gt; alter (elem Ni), vdw=0.83
 PyMOL&gt; rebuild</code></pre>

<div class="figure"><img src="/img/Ni.png" alt="" />
 <p class="caption"></p></div>
</article>
<article>
  <header>
    <h2><a href='/blog/2014/12/10/script-to-show-methyls-as-spheres-in-pymol/'>Script to show methyls as spheres in Pymol</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-12-10" pubdate="true">2014-12-10</time> :: <span class="tags"><a href="/tags/pymol.html">pymol</a></span></p>
  </header>

<p>Here is a pymol script to show methyl groups on your protein as spheres, color-coded by residue type.</p>

<p>Download the script <a href="https://raw.githubusercontent.com/betainverse/methyls/master/pymol/showmethyls.py">here</a>.</p>

<p>Usage: Load the script into pymol using the run&hellip; item in the pymol File menu to select this script or load it from the Pymol command line, then actually run it using the showmethyls command:</p>

<pre><code> PyMOL&gt; run /home/userName/path/toscript/showmethyls.py
 PyMOL&gt; showmethyls</code></pre>

<p>Note that you can use the cd command on the pymol command line as you would in bash/tcsh, if you don&rsquo;t remember the exact path to the script by heart.</p>

<div class="figure"><img src="/img/spherelegend.png" alt="" />
 <p class="caption"></p></div>

<div class="figure"><img src="/img/methyls2.png" alt="" />
 <p class="caption"></p></div>
</article>
<article>
  <header>
    <h2><a href='/blog/2014/10/30/simple-linux-tutorials/'>simple linux tutorials</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-10-30" pubdate="true">2014-10-30</time> :: <span class="tags"><a href="/tags/linux.html">linux</a></span></p>
  </header>

<ul>
 <li><a href="http://pgbovine.net/command-line-tutorial.htm">http://pgbovine.net/command-line-tutorial.htm</a></li></ul>
</article>
<article>
  <header>
    <h2><a href='/blog/2014/10/14/coloring-methyls-in-pymol/'>Coloring Methyls in Pymol</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-10-14" pubdate="true">2014-10-14</time> :: <span class="tags"><a href="/tags/pymol.html">pymol</a></span></p>
  </header>

<p>This post expands on <a href="http://betainverse.github.io/blog/2014/10/13/pymol-color-by-data/">yesterday&rsquo;s</a> post that was focused on coloring protein structures by residue in Pymol.</p>

<ol>
 <li>
  <p>Your data should be in a tab-delimited text file, formatted like this:</p>
  <pre><code>2	ALA	CB	0.03416350
8	ILE	CD1	0.43143940
13	LEU	CD1	0.50597498</code></pre></li>
 <li>Make sure you have downloaded data2bfactor.py from <a href="http://pldserver1.biochem.queensu.ca/~rlc/work/pymol/">http://pldserver1.biochem.queensu.ca/~rlc/work/pymol/</a>.</li>
 <li>Open pymol &amp; load your protein structure.</li>
 <li>
  <p>Run any scripts you want to use:</p>
  <pre><code>PyMOL&gt; run ~/scripts/data2bfactor.py
PyMOL&gt; run ~/scripts/spectrumany.py</code></pre></li>
 <li>
  <p>You may want to set all the b-factor data for your protein to &ndash;1 or to some other number beforehand, because any residues not mentioned in your data file will retain their original crystallographic B-factor:</p>
  <pre><code>PyMOL&gt; alter MyProtein, b=-1</code></pre></li>
 <li>
  <p>Define a selection of all your methyls:</p>
  <pre><code>PyMOL&gt; select MyMethyls, (resn ALA and name CB)+(resn ILE and name CD1)+(resn LEU and name CD1+CD2)+(resn MET+MSE and name CE)+(resn VAL and name CG1+CG2)+(resn THR and name CG2)</code></pre></li>
 <li>
  <p>Display things nicely:</p>
  <pre><code>PyMOL&gt; hide lines
PyMOL&gt; show cartoon
PyMOL&gt; dss
PyMOL&gt; color gray80, MyProtein
PyMOL&gt; show spheres, MyMethyls
PyMOL&gt; show sticks, resn ALA+ILE+LEU+VAL+MET and not (name c,o,n)</code></pre></li>
 <li>
  <p>Now load your data onto your selection using the data2b_atom function defined within data2bfactor.py.</p>
  <pre><code>PyMOL&gt; data2b_atom MyMethyls, /Users/username/Documents/datafile.txt</code></pre></li>
 <li>
  <p>Apply the color gradient:</p>
  <pre><code>PyMOL&gt; spectrumany b, red yellow, methyls</code></pre></li>
 <li>Ray trace, save the image!</li></ol>

<div class="figure"><img src="/img/methyls.png" alt="" />
 <p class="caption"></p></div>
</article>
<article>
  <header>
    <h2><a href='/blog/2014/10/13/pymol-color-by-data/'>Pymol Color By Data</a></h2>
    <p class='date-and-tags'>
<time datetime="2014-10-13" pubdate="true">2014-10-13</time> :: <span class="tags"><a href="/tags/pymol.html">pymol</a></span></p>
  </header>

<p>This post expands on <a href="http://kpwu.wordpress.com/2007/11/27/pymol-example-coloring-surface-by-b-factor/">info</a> from a few other <a href="http://kpwu.wordpress.com/2012/09/11/pymol-custom-spectrum-colors/">blog</a> posts.</p>

<ol>
 <li>
  <p>Your data should be in a tab-delimited text file, formatted like this:</p>
  <pre><code>3	0.677985
4	0.794402
5	0.972709</code></pre></li>
 <li>
  <p>You may need to remove extra columns and/or headerscan cause problems. If you exported your file from excel, you may have to change your line break format. From a mac, you might see a bunch of ^M characters in your file. You can use a terminal to do one of these commands to fix the file:</p>
  <pre><code>tr '\r' '\n' &lt; macfile.txt &gt; unixfile.txt
tr -d '\r' &lt; windowsfile.txt &gt; unixfile.txt</code></pre></li>
 <li>Save this data file to a known location, let&rsquo;s say /Users/username/Documents/datafile.txt. It might make things easier to avoid having spaces in file or folder names. It seems to be important to use the absolute filename, rather than using shortcuts like <code>~</code>.</li>
 <li>Download data2bfactor.py from <a href="http://pldserver1.biochem.queensu.ca/~rlc/work/pymol/">http://pldserver1.biochem.queensu.ca/~rlc/work/pymol/</a>.</li>
 <li>Download spectrumany.py from <a href="http://pymolwiki.org/index.php/Spectrumany">http://pymolwiki.org/index.php/Spectrumany</a>.</li>
 <li>Save these scripts to a known location, let&rsquo;s say ~/scripts/.</li>
 <li>Open pymol &amp; load your protein structure.</li>
 <li>
  <p>Run the scripts you downloaded using the PyMOL&gt; commandline:</p>
  <pre><code>PyMOL&gt; run ~/scripts/data2bfactor.py
PyMOL&gt; run ~/scripts/spectrumany.py</code></pre></li>
 <li>
  <p>Make a named selection for the set of residues you want to color:</p>
  <pre><code>PyMOL&gt; select MyChainA, 3TGN and chain A and not resn Zn</code></pre></li>
 <li>
  <p>You may want to set all the b-factor data for your selection to 0 or to some other number beforehand, because any residues not mentioned in your data file will retain their original crystallographic B-factor:</p>
  <pre><code>PyMOL&gt; alter MyChainA, b=0</code></pre></li>
 <li>
  <p>Now load your data onto your selection using the data2b_res function defined within data2bfactor.py:</p>
  <pre><code>PyMOL&gt; data2b_res MyChainA, /Users/username/Documents/datafile.txt</code></pre></li>
 <li>
  <p>Now color by B-factor. If you are lucky and the color gradient you want is already included in Pymol, you can use the function spectrum:</p>
  <pre><code>PyMOL&gt; spectrum b, rainbow, MyChainA, minimum=0.6, maximum=1</code></pre></li>
 <li>You will need to play around with minimum and maximum to optimize the image according to your data.</li>
 <li>
  <p>If you want a color gradient not already defined in Pymol, you will need to use the spectrumany command:</p>
  <pre><code>PyMOL&gt; spectrumany b, red gray80, MyChainA, minimum=0.6,maximum=1</code></pre></li>
 <li>
  <p>Now use the Display menu to make sure the Background is not set to be opaque, and ray trace the image:</p>
  <pre><code>PyMOL&gt; ray</code></pre></li>
 <li>Save your image and your session!</li></ol>

<div class="figure"><img src="/img/Zn_hNOE.png" alt="" />
 <p class="caption"></p></div>
</article>
<footer>
 <ul class="pagination">
  <li class="disabled"><a href="#">
    <quote>&larr;</quote></a></li>
  <li class="active"><a href="/index.html">1</a></li>
  <li class="disabled"><a href="#">
    <quote>&rarr;</quote></a></li></ul></footer>
        </div>
      </div>
      <footer>
        <hr />
        <p><a href="https://twitter.com/racketlang"
              class="twitter-follow-button"
              data-show-count="false"
              data-lang="en">
             "Follow RacketLang"
           </a>
           <script type="text/javascript">
             !function(d,s,id){
                 var js,fjs=d.getElementsByTagName(s)[0];
                 if(!d.getElementById(id)){
                     js=d.createElement(s);
                     js.id=id;
                     js.src="//platform.twitter.com/widgets.js";
                     fjs.parentNode.insertBefore(js,fjs);
                 }
             }(document,"script","twitter-wjs");
           </script></p>
        <p>Site generated
        by <a href="https://github.com/greghendershott/frog">Frog</a>,
        the <strong>fr</strong>ozen bl<strong>og</strong> tool.</p>
        <p>Using <a href="http://twitter.github.com/bootstrap/index.html">Bootstrap</a>.</p>
        <p><em>Your legal notice here</em>.</p>
      </footer>
    </div>
    <!-- </body> JS -->
    <script type="text/javascript" src="//code.jquery.com/jquery.js"></script>
    <script type="text/javascript" src="/js/bootstrap.min.js"></script>
  </body>
</html>